src/main/java/io/pravega/connectors/hadoop/PravegaInputRecordReader.java

```diff
@@ -61,13 +62,25 @@ protected PravegaInputRecordReader(BatchClientFactory externalClientFactory) {
      */
     @Override
     public void initialize(InputSplit split, TaskAttemptContext context) throws IOException, InterruptedException {
-        initialize(split, context.getConfiguration());
-    }
-
-    public void initialize(InputSplit split, Configuration conf) throws IOException, InterruptedException {
         this.split = (PravegaInputSplit) split;
+        Configuration conf = context.getConfiguration();
+
+        final URI controllerURI = Optional.ofNullable(conf.get(PravegaConfig.INPUT_URI_STRING)).map(URI::create).orElseThrow(() ->
+                new IOException("The Pravega controller URI must be configured (" + PravegaConfig.INPUT_URI_STRING + ")"));
+
+        PravegaClientConfig pravegaClientConfig = PravegaClientConfig.fromDefaults();
```

Comment:
Code from line 71 to 82 is repeating here. It was also there in PravegaInputFormat. I'd suggest abstracting it out and reusing in both these places.

Reply:
reorganized code to use helper class to derive the configuration instance